BlackJack/Game.cs

```diff
@@ -32,95 +35,148 @@ public void Start()
             _player.Hand.AddCard(_deck.GetNextCard());
             _player.Hand.AddCard(_deck.GetNextCard());
 
-            DealerPlayer dealer = new DealerPlayer();
-            dealer.Hand = new Hand();
-            dealer.Hand.AddCard(_deck.GetNextCard());
+            _dealer.Hand = new Hand();
+            _dealer.Hand.AddCard(_deck.GetNextCard());
 
             OnGameStart(new OnGameStartArgs()
             {
                 Player = _player,
-                Dealer = dealer
+                Dealer = _dealer
             });
 
-            while (!_player.Hand.IsBust)
+            List<Card> cards = _player.Hand.GetCards();
+            if (cards[0].Face == cards[1].Face)
             {
-                TurnAction turnAction = OnGameTurn(new OnGameTurnArgs()
+                SplitAction splitAction = OnGameSplit(new OnGameSplitArgs()
                 {
                     Player = _player
                 });
 
-                if (turnAction == TurnAction.Hit)
+                if (splitAction == SplitAction.Yes)
                 {
+                    Hand splitHand = _player.Hand.Split();
                     _player.Hand.AddCard(_deck.GetNextCard());
-                    OnGameHit(new OnGameHitArgs()
-                    {
-                        Player = _player
-                    });
-                }
-                else if (turnAction == TurnAction.Stay)
-                {
-                    OnGameStay(new OnGameStayArgs()
-                    {
-                        Player = _player
-                    });
-                    break;
+                    splitHand.AddCard(_deck.GetNextCard());
+                    _player.SplitHand = splitHand;
                 }
-                else
+            }
+
+            bool isFirstHandAce = cards[0].Face == Face.Ace;
+            List<Card> splitHandCards = _player.SplitHand.GetCards();
```

Comment:
The hand might not be split at this point, will cause a NullReferenceException (_player.SplitHand == null)